src/modules/mc_pos_control/mc_pos_control_main.cpp

```diff
@@ -583,47 +587,34 @@ MulticopterPositionControl::run()
 	_home_pos_sub = orb_subscribe(ORB_ID(home_position));
 	_traj_wp_avoidance_sub = orb_subscribe(ORB_ID(vehicle_trajectory_waypoint));
 
+	// get initial values for all parameters and subscribtions
 	parameters_update(true);
-
-	// get an initial update for all sensor and status data
 	poll_subscriptions();
 
-	// We really need to know from the beginning if we're landed or in-air.
-	orb_copy(ORB_ID(vehicle_land_detected), _vehicle_land_detected_sub, &_vehicle_land_detected);
-
-	hrt_abstime t_prev = 0;
-
 	// Let's be safe and have the landing gear down by default
 	_att_sp.landing_gear = vehicle_attitude_setpoint_s::LANDING_GEAR_DOWN;
 
-	// wakeup source
-	px4_pollfd_struct_t fds[1];
-
-	fds[0].fd = _local_pos_sub;
-	fds[0].events = POLLIN;
+	// setup file descriptor to poll the local position as loop wakeup source
+	px4_pollfd_struct_t poll_fd = {.fd = _local_pos_sub};
```

Comment:
This leaves other fields in poll_fd uninitialized. You either need to set all fields in the braces, or do a single '{}', then set only fd and events after.

Reply:
I beg to differ:
> All members that are not initialized explicitly are initialized implicitly the same way as objects that have static storage duration.

https://en.cppreference.com/w/c/language/struct_initialization

Reply:
Thanks. TIL